DXMainClient/Domain/Multiplayer/Map.cs

```diff
@@ -107,72 +120,96 @@ public Map(string path, bool official)
         /// <summary>
         /// The file name of the preview image.
         /// </summary>
+        [JsonProperty]
         public string PreviewPath { get; private set; }
 
         /// <summary>
         /// If set, this map cannot be played on Skirmish.
         /// </summary>
+        [JsonProperty]
         public bool MultiplayerOnly { get; private set; }
 
         /// <summary>
         /// If set, this map cannot be played with AI players.
         /// </summary>
+        [JsonProperty]
         public bool HumanPlayersOnly { get; private set; }
 
         /// <summary>
         /// If set, players are forced to random starting locations on this map.
         /// </summary>
+        [JsonProperty]
         public bool ForceRandomStartLocations { get; private set; }
 
         /// <summary>
         /// If set, players are forced to different teams on this map.
         /// </summary>
+        [JsonProperty]
         public bool ForceNoTeams { get; private set; }
 
         /// <summary>
         /// The name of an extra INI file in INI\Map Code\ that should be
         /// embedded into this map's INI code when a game is started.
         /// </summary>
+        [JsonProperty]
         public string ExtraININame { get; private set; }
 
         /// <summary>
         /// The game modes that the map is listed for.
         /// </summary>
+        [JsonProperty]
         public string[] GameModes;
 
         /// <summary>
         /// The forced UnitCount for the map. -1 means none.
         /// </summary>
+        [JsonProperty]
         int UnitCount = -1;
 
         /// <summary>
         /// The forced starting credits for the map. -1 means none.
         /// </summary>
+        [JsonProperty]
         int Credits = -1;
 
+        [JsonProperty]
         int NeutralHouseColor = -1;
 
+        [JsonProperty]
         int SpecialHouseColor = -1;
 
+        [JsonProperty]
         int Bases = -1;
 
+        [JsonProperty]
         string[] localSize;
 
+        [JsonProperty]
         string[] actualSize;
 
-        IniFile mapIni;
+        private IniFile CustomMapIni;
```

Comment:
@Rampastring is this OK with our code style?

Reply:
Private fields should be in lowercase.

Reply:
Resolved